src/Physics/HadronTransport/INukeHadroData2025.cxx

```diff
@@ -0,0 +1,1900 @@
+  delete fXSecGamn_fs;
+  delete fXSecGamN_Tot;
+
+  // N+A x-section splines
+  delete fFracPA_Tot;
+  //  delete fFracPA_Elas;
+  delete fFracPA_Inel;
+  delete fFracPA_CEx;
+  delete fFracPA_Abs;
+  delete fFracPA_PiPro;
+  delete fFracNA_Tot;
+  //  delete fFracNA_Elas;
+  delete fFracNA_Inel;
+  delete fFracNA_CEx;
+  delete fFracNA_Abs;
+  delete fFracNA_PiPro;
+
+  delete fFracPA_Cmp;  //  cmp - add support later
+  delete fFracNA_Cmp;
+
+  // hN data
+  delete fhN2dXSecPP_Elas;
+  delete fhN2dXSecNP_Elas;
+  delete fhN2dXSecPipN_Elas;
+  delete fhN2dXSecPi0N_Elas;
+  delete fhN2dXSecPimN_Elas;
+  delete fhN2dXSecKpN_Elas;
+  delete fhN2dXSecKpP_Elas;
+  delete fhN2dXSecPiN_CEx;
+  delete fhN2dXSecPiN_Abs;
+  delete fhN2dXSecGamPi0P_Inelas;
+  delete fhN2dXSecGamPi0N_Inelas;
+  delete fhN2dXSecGamPipN_Inelas;
+  delete fhN2dXSecGamPimP_Inelas;
+  delete fhN2dXSecKpN_CEx;
+  
+ // delete TPipA_Tot;
+  delete TfracPipA_Abs;
+  delete TfracPipA_CEx;
+//  delete TfracPipA_Elas;
+  delete TfracPipA_Inelas;
+  delete TfracPipA_PiPro;
+    
+    delete TPipA_Tot;
+    delete TPipA_Abs;
+    delete TPipA_CEx;
+    delete TPipA_Inelas;
+    delete TPipA_PiPro;
+    
+  
+  // K+A x-section fraction splines
+  delete fFracKA_Tot;
+  delete fFracKA_Elas;
+  delete fFracKA_CEx;
+  delete fFracKA_Inel;
+  delete fFracKA_Abs;
+  
+}
+//____________________________________________________________________________
+INukeHadroData2025 * INukeHadroData2025::Instance()
+{
+  if(fInstance == 0) {
+    LOG("INukeData", pINFO) << "INukeHadroData2025 late initialization";
+    static INukeHadroData2025::Cleaner cleaner;
+    cleaner.DummyMethodAndSilentCompiler();
+    fInstance = new INukeHadroData2025;
+  }
+  return fInstance;
+}
+//____________________________________________________________________________
+void INukeHadroData2025::LoadCrossSections(void)
+{
+// Loads hadronic x-section data
+
+  //-- Get the top-level directory with input hadron cross-section data
+  //   (search for $GINUKEHADRONDATA or use default location)
+  string data_dir = (gSystem->Getenv("GINUKEHADRONDATA")) ?
+             string(gSystem->Getenv("GINUKEHADRONDATA")) :
+             string(gSystem->Getenv("GENIE")) + string("/data/evgen/intranuke");
+
+  LOG("INukeData", pINFO)
+      << "Loading INTRANUKE hadron data from: " << data_dir;
+
+  //-- Build filenames
+
+  string datafile_NN   = data_dir + "/tot_xsec/intranuke-xsections-NN2014.dat";
+  string datafile_pipN = data_dir + "/tot_xsec/intranuke-xsections-pi+N.dat";
+  string datafile_pi0N = data_dir + "/tot_xsec/intranuke-xsections-pi0N.dat";
+  string datafile_NA   = data_dir + "/tot_xsec/intranuke-fractions-NA2016.dat";
+  string datafile_KA   = data_dir + "/tot_xsec/intranuke-fractions-KA.dat";
+  string datafile_gamN = data_dir + "/tot_xsec/intranuke-xsections-gamN.dat";
+  string datafile_kN   = data_dir + "/tot_xsec/intranuke-xsections-kaonN2018.dat";
+
+  //-- Make sure that all data files are available
+
```

Comment:
Just my usual note that `assert()` in "production" code (vs. debug code) is a no-op and won't catch errors the same way it does in debug code.   I'm not saying these are useless, just be aware.